render/gles2/renderer.c

```diff
@@ -116,7 +116,7 @@ static bool gles2_render_texture_with_matrix(struct wlr_renderer *wlr_renderer,
 	struct wlr_gles2_renderer *renderer =
 		gles2_get_renderer_in_context(wlr_renderer);
 	struct wlr_gles2_texture *texture =
-		get_gles2_texture_in_context(wlr_texture);
+		gles2_get_texture(wlr_texture);
```

Comment:
Can you try adding back this assertion? I'd like to assert that the texture has been created in the same context as the renderer.